enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/analytics.ts

```diff
@@ -0,0 +1,205 @@
+  AUTH_TYPES,
+  DefaultValues,
+  EmbeddedAnalyticsJsEventSchema,
+} from "metabase-types/analytics/embedded-analytics-js";
+
+import type { MetabaseEmbedElement } from "./embed";
+
+const DEFAULT_VALUES: DefaultValues = {
+  dashboard: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L119} */
+    drills: true,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/embedding-sdk-bundle/components/public/dashboard/SdkDashboard.tsx#L129} */
+    with_downloads: false,
+    /** @see {@link https://github.com/metabase/metabase/blob/master/enterprise/frontend/src/embedding-sdk-bundle/components/public/dashboard/SdkDashboard.tsx#L127} */
+    with_title: true,
+  },
+  question: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L149} */
+    drills: true,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/embedding-sdk-bundle/components/public/SdkQuestion/SdkQuestion.tsx#L129} */
+    with_downloads: false,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L145} */
+    with_title: true,
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L157} */
+    is_save_enabled: false,
+  },
+  exploration: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx#L157} */
+    is_save_enabled: false,
+  },
+  browser: {
+    /** @see {@link https://github.com/metabase/metabase/blob/7aa3f7fed11113ed32901aad4e4227be68cff78f/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/MetabaseBrowser.tsx#L39} */
+    read_only: true,
+  },
+};
+
+const SECOND = 1000;
+
+/**
+ * Because the way it's designed to be consumed when each embed is being registered. There could be multiple embed in a page, so we want to debounce and only call this once.
+ */
+export const debouncedReportAnalytics = _.debounce(
+  reportAnalyticsOnce,
+  1 * SECOND,
+);
+
+/**
+ * This is just a safe-guard to ensure we only call this once for whatever reason.
+ */
+let isAnalyticsReported = false;
+function reportAnalyticsOnce(activeEmbeds: Set<MetabaseEmbedElement>) {
+  if (!isAnalyticsReported) {
+    isAnalyticsReported = true;
+    const [firstEmbed] = activeEmbeds;
```

Comment:
`firstEmbed` will always be there because we call this function after registering them.

https://github.com/metabase/metabase/blob/4aa4e5b68304ed61d3b1a8a32986743fd5c565b4/enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.ts#L83-L84